pkg/network/server.go

```diff
@@ -138,6 +138,8 @@ type (
 		stateSync StateSync
 
 		log *zap.Logger
+
+		shutdownOnce sync.Once
```

Comment:
To match the overall pattern of NeoGo services, port the approach with `started` atomic bool variable:
https://github.com/nspcc-dev/neo-go/blob/51ac153a7b009f8f69117ed987038a08dc78f261/pkg/services/rpcsrv/server.go#L145